### PR TITLE
[16.0][OU-ADD] hr_fleet

### PR DESCRIPTION
**Pull request by nguyenvietlam0640**

### CODE REVIEW

**Review by legalsylvain — APPROVED** [review]

**Comment by legalsylvain** [comment]
/ocabot migration hr_fleet

:+1: 

Note : you can run ``pre-commit run -a`` before commiting to be sure CI is green.

Could you squach commits ? 

<dependency>Depends on : 

- fleet : #3947</dependency>

**Comment by nguyenvietlam0640** [comment]
> /ocabot migration hr_fleet
> 
> +1
> 
> Note : you can run `pre-commit run -a` before commiting to be sure CI is green.
> 
> Could you squach commits ?

Done

File: openupgrade_scripts/scripts/hr_fleet/16.0.1.0/upgrade_analysis_work.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---Models in module 'hr_fleet'---
+---Fields in module 'hr_fleet'---
+hr_fleet     / fleet.vehicle.assignation.log / driver_employee_id (many2one) : is now stored
+hr_fleet     / fleet.vehicle.assignation.log / driver_employee_id (many2one) : not related anymore
+
+# NOTHING TO DO: field changed from related field to compute field
```
Comment (hoangtiendung070797) [inline openupgrade_scripts/scripts/hr_fleet/16.0.1.0/upgrade_analysis_work.txt]: @nguyenvietlam0640 i think need migration fill value for it

File: openupgrade_scripts/scripts/hr_fleet/16.0.1.0/pre-migration.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            SET driver_employee_id = %s
+            WHERE vehicle_id = %s
+            """,
+            (vehicle[1], vehicle[0]),
+        )
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    create_drive_employee(env)
```
Comment (hoangtiendung070797) [inline openupgrade_scripts/scripts/hr_fleet/16.0.1.0/pre-migration.py]:
```suggestion
@openupgrade.migrate()
def migrate(env, version):
    openupgrade.logged_query(
        env.cr,
        """
        ALTER TABLE fleet_vehicle_assignation_log
        ADD COLUMN IF NOT EXISTS driver_employee_id integer
        """,
    )
    openupgrade.logged_query(
        env.cr,
        """
        UPDATE fleet_vehicle_assignation_log AS fval
        SET driver_employee_id = v.driver_employee_id
        FROM fleet_vehicle AS v
        WHERE v.id = fval.vehicle_id
        """,
    )
```

Comment (legalsylvain) [reply openupgrade_scripts/scripts/hr_fleet/16.0.1.0/pre-migration.py]: please keep a dedicated function to populate driver_employee_id. the migrate function should only contains call to others function.

Comment (hoangtiendung070797) [reply openupgrade_scripts/scripts/hr_fleet/16.0.1.0/pre-migration.py]: > please keep a dedicated function to populate driver_employee_id. the migrate function should only contains call to others function.

@legalsylvain Ok. I got it. Thank you

**Review by victoralmau — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**